### PR TITLE
Fix/stasforecast 1 6 release

### PR DESCRIPTION
**Pull request by madtoinou**
### Summary

- Stasforecast fixed a bug allowing `AutoETS` model to use the `"A*M"` and `"AM*"` combination when passing `"Z"` as parameter for the trend and/or seasonality. Darts' unittest was using `"AAZ"`, effectively relying on an invalid error value (which was increased from 4.2 to 7.3 in this PR)
- Updated the links to statsforecast documentation as they modified the repository organization.
- Updated the documentation of all the models to make them more homogeneous and included a little note about conformal prediction (introduced in their 1.6.0 release).

### Other Information
- The fitted parameters (values of the "Z" placeholder after training) can be found using `model.model.model_["components"]`

### CODE REVIEW

File: darts/models/forecasting/sf_auto_arima.py [hunk 1]
```diff
@@ -32,15 +32,11 @@ def __init__(
...
-        for the documentation of the arguments.
+        <https://nixtla.github.io/statsforecast/src/core/models.html#autoarima>`_
+        for the exhaustive documentation of the arguments.
 
         Parameters
         ----------
-        autoarima_args
-            Positional arguments for ``statsforecasts.models.AutoARIMA``.
-        autoarima_kwargs
-            Keyword arguments for ``statsforecasts.models.AutoARIMA``.
```
Comment (dennisbader) [inline darts/models/forecasting/sf_auto_arima.py:43]: I would revert the changes to this and the "note" section and do it similarly as we have it for TorchForecastingModels and the `**tfm_kwargs` arguments

File: darts/models/forecasting/sf_auto_ces.py [hunk 2]
```diff
@@ -18,17 +18,27 @@ def __init__(self, *autoces_args, **autoces_kwargs):
...
+            * "S" for simple simple seasonality (lagged CES),
+            * "P" for partial seasonality (without complex part),
+            * "F" for full seasonality (lagged CES with both real and complex seasonality).
+
+            Furthermore, the character "Z" is a placeholder telling statsforecast
+            to search for the best parameter. Default: "Z".
+
+        .. note::
+            Positional and keyword arguments can be used to pass additional parameters to statsforecast's `AutoCES`
+            constructor, such as ``prediction_intervals`` to compute the conformal prediction intervals.
```
Comment (dennisbader) [inline darts/models/forecasting/sf_auto_ces.py:41]: I think it's simpler if we leave it as before. Otherwise, we have to maintain this ourselves, and check whether it still complies with newer versions of statsforecast.

File: darts/tests/models/forecasting/test_local_forecasting_models.py [hunk 3]
```diff
@@ -56,7 +56,7 @@
     (StatsForecastAutoARIMA(season_length=12), 4.6),
     (StatsForecastAutoTheta(season_length=12), 5.5),
     (StatsForecastAutoCES(season_length=12, model="Z"), 7.3),
-    (StatsForecastAutoETS(season_length=12, model="AAZ"), 4.1),
+    (StatsForecastAutoETS(season_length=12, model="AAZ"), 7.3),
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_local_forecasting_models.py:59]: just to check, was statsforecast previously wrongly using a different `model` for "AAZ", or was it just a bug in general?
If we the former applies, could we use that other model to achieve the 4.1 error?

**Review by dennisbader — CHANGES_REQUESTED** [review]
Thanks a lot for checking this @madtoinou , glad that you found out where the issue is coming from 🚀 
I would suggest to revert the changes to the parameter descriptions as otherwise, we will have to maintain the descriptions in the future.

File: darts/tests/models/forecasting/test_local_forecasting_models.py (hunk 3 above)
Comment (madtoinou) [reply darts/tests/models/forecasting/test_local_forecasting_models.py:59]: During the parameter optimization for the placeholder "Z" (seasonality state), the `"AAM"` combination was tested (which is apparently not allowed, I did not investigate why) and they fixed this bug. For the record, with 1.5.0, passing explicitly `"AAM"` would raise an error.

Comment (dennisbader) [reply darts/tests/models/forecasting/test_local_forecasting_models.py:59]: sounds good, thanks!

**Review by dennisbader — APPROVED** [review]
